datastore/datastore_test.go

```diff
@@ -3484,15 +3584,15 @@ func TestDeferredMissing(t *testing.T) {
 }
 
 func TestGetWithNilKey(t *testing.T) {
-	client := &Client{}
+	client := &Client{readSettings: &readSettings{}}
```

Comment:
This should be done in the `NewClient` initializer.

Reply:
The `readSettings` are now set in `NewClient()`. However, refactoring all of the unit tests to use `NewClient()` is beyond the scope of this PR.